client/asset/eth/multirpc.go

```diff
@@ -1059,18 +1059,21 @@ func (m *multiRPCClient) withAll(
 ) error {
 
 	var atLeastOne bool
-	var errs []error
-	for _, p := range m.nonceProviderList() {
+	var errs []string
+	providers := m.nonceProviderList()
+	for _, p := range providers {
 		if p.failed() {
 			continue
 		}
+
 		ctx, cancel := context.WithTimeout(ctx, defaultRequestTimeout)
 		err := f(ctx, p)
 		cancel()
 		if err == nil {
 			atLeastOne = true // return nil err unless a later "propagated" error says to
 			continue
 		}
+		errs = append(errs, err.Error())
```

Comment:
Actually, you can't append this here because they might have a filter that tells us to discard it.

Reply:
So which is preferable in this case? Continue login for each provider at L1094 or add to `errs`, remove that log and return all errs to caller?

Reply:
I think you just need to append the error in the `else` block where we are doing the `log.Warn`. That's the only change that was needed in this method, from what I can tell.

Reply:
Yh, but then I wonder is we should still keep the log?